From a578200ea6bd936ae97ed7830c40cd340fb8e4c1 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:57 -0400
Subject: [fs] btrfs: fix compile warnings on on avr32 platform

Message-id: <1407179109-13714-291-git-send-email-sandeen@redhat.com>
Patchwork-id: 87151
O-Subject: [RHEL7.1 PATCH 290/422] Btrfs: fix compile warnings on on avr32 platform
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit e4fbaee29272533a242f117d18712e2974520d2c
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Fri Apr 11 18:32:25 2014 +0800

    Btrfs: fix compile warnings on on avr32 platform

    fs/btrfs/scrub.c: In function 'get_raid56_logic_offset':
    fs/btrfs/scrub.c:2269: warning: comparison of distinct pointer types lacks a cast
    fs/btrfs/scrub.c:2269: warning: right shift count >= width of type
    fs/btrfs/scrub.c:2269: warning: passing argument 1 of '__div64_32' from incompatible pointer type

    Since @rot is an int type, we should not use do_div(), fix it.

    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/scrub.c b/fs/btrfs/scrub.c
index aee909f..68a5a26 100644
--- a/fs/btrfs/scrub.c
+++ b/fs/btrfs/scrub.c
@@ -2266,7 +2266,7 @@ static int get_raid56_logic_offset(u64 physical, int num,
   rot = do_div(stripe_nr, map->num_stripes);
   /* calculate which stripe this data locates */
   rot += i;
-  stripe_index = do_div(rot, map->num_stripes);
+  stripe_index = rot % map->num_stripes;
   if (stripe_index == num)
    return 0;
   if (stripe_index < num)
-- 
1.7.1